### ISSUE TITLE
Is there a way to get live videos? 

### ISSUE THREAD

**Issue by johndpope** [isaackogan/TikTokLive#87]
**Is your feature request related to a problem? Please describe.**
tiktok.com/live

**Describe the solution you'd like**
a json of videos / titles
https://pull-hls-f16-va01.tiktokcdn.com/stage/stream-2994245018867728457_ld/index.m3u8
https://pull-f5-va01.tiktokcdn.com/stage/stream-2994245018867728457_hd.flv

**Describe alternatives you've considered**
soup / viewing source
https://github.com/ychen1214/python/blob/f84f78ae7fdacf7db4ebc5fff4d98f0da2c73b0b/beautifulsoup/soupinaction.py

**Additional context**
want to show in an app

https://www.tiktok.com/@dappestcontrol/live
if you access via a username - there's a list
<img width="887" alt="Screenshot 2022-11-10 at 5 29 00 pm" src="https://user-images.githubusercontent.com/289994/201016999-3bfec0bc-861c-4cf9-a964-b5cece435923.png">

**Comment by johndpope** [comment]
```shell

curl 'https://webcast.tiktok.com/webcast/feed/?aid=1988&app_language=en&app_name=tiktok_web&battery_info=0.95&browser_language=en-GB&browser_name=Mozilla&browser_online=true&browser_platform=MacIntel&browser_version=5.0%20%28Macintosh%3B%20Intel%20Mac%20OS%20X%2010_15_7%29%20AppleWebKit%2F537.36%20%28KHTML%2C%20like%20Gecko%29%20Chrome%2F107.0.0.0%20Safari%2F537.36&channel=tiktok_web&channel_id=86&content_type=0&cookie_enabled=true&device_id=7134872442754188802&device_platform=web_pc&focus_state=false&from_page=user&hidden_banner=true&history_len=16&is_fullscreen=false&is_page_visible=true&max_time=0&os=mac&priority_region=AU&referer=&region=AU&req_from=pc_web_suggested_host&screen_height=982&screen_width=1512&tz_name=Australia%2FSydney&webcast_language=en&msToken=qqL_LL1SgBvIGmZAf9LW7luKyQhm7WHYCW0g4OZ7AqZq2CESkHZ0EKfNQoHWxPVn0XXMCo93CAPunyzFH3csp0iEbotTFEsxw0mDDmyxU9RBbHn11Rqywbs0iyDsoT9so0BvyJJaQliJM74=&X-Bogus=DFSzswVuGnJAN9jsS0il1p7TlqSA&_signature=_02B4Z6wo00001W0AuAwAAIDC-dbjA01oJeVtALyAADgw06' \
  -H 'authority: webcast.tiktok.com' \
  -H 'accept: */*' \
  -H 'accept-language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'cookie: _ttp=2CihgY1ygHg9OrojPiipd9WeTgX; passport_csrf_token=98536af196df0de0bda32572ae7939a8; passport_csrf_token_default=98536af196df0de0bda32572ae7939a8; cmpl_token=AgQQAPPdF-RO0ozd1EICPR08-WWxnJOJf4B0YMTRuw; uid_tt=146da7d8c5724930d8e2630a027b2bbd06a20f460f778e5142e6835d110db00b; uid_tt_ss=146da7d8c5724930d8e2630a027b2bbd06a20f460f778e5142e6835d110db00b; sid_tt=ccee2b81e4750e6ea2143ce3da193ada; sessionid=ccee2b81e4750e6ea2143ce3da193ada; sessionid_ss=ccee2b81e4750e6ea2143ce3da193ada; store-idc=alisg; store-country-code=au; store-country-code-src=uid; tt-target-idc=alisg; tt-target-idc-sign=ZFMl3e6uDrGUqsROQcgB29DAcmiO-YJgkcxowQKCyNiaUHklxupyS5-XBZH7GuBp2O8SJEUwCkJGiiqRQ9nx5ra7rSHerFk_TBe-p6IAEwN8-w09AwDg9bnYkCsKNGdDef5LGqmLBOMZQxEX1OGt3zQYP5yDuXhABEjMymldyDcKlB_QNizqZItlWrA6KIvHqkgvVydMlJaSuUIJhw3Mg2i4kXJ55cwFWSxGrTYhXmX3YFF8czGHXVnUcpXaOxQQCnbpLFMRnwvNp9TLM0dfIytDMlJYNhIdMScQgcxXdESChe_7ByO0k_jRHnASylHHKHAbCT5w2rBYRb0AdioksujA4Ibu8mohVEli5UOo5kRFiZ6rvg18rzYmMzwLecAw8TZ9H8siAGiMBbY1Hd9ws8Et8H2v_Vn8scoFUQKFBygrT-Jf8vxPawZm4fce91Y_AlyQQqvDe5Si3Y6IWFT9sWpyF16A8GatYZsgiZLxmFI10QjYrNLuCQFtSkEsfBNY; tt_csrf_token=jEJBLjOD-FwitMB_B-G5WyYM40ZSslJAodFw; _abck=70BC0444C961E9B31B11D5B29C4F0759~-1~YAAQ9bEHyuK7Lz6EAQAA2wEaYAjaX/I4THwHn9lWtNk0TXkCgVXx5+vAqzuqn/Xb4VjsI6nA0EWJGn3wuNVf7OuHQStz/ba2xXaKldHhk/YNeELcPplr6ZLm04qZE+7DyUVotJItN0OqCTyNOjOb3CQUPNJ51jVgsGfQiXj2WOirVh8usVSq/jC0RlMyGIFJpKX6UQ3xMgTO0EmbqkE8WnEn9mnRWCV25Dfs0lRIOMntCtqPhjGCYX3RZMDktBbnSEKECkCxPvM4Q1PpsT7gXAcGa6dliL0vOFeErNF82V4FMFvQQabVGtLKNsb+dGxdQqsBJJGNazbnyQ4emu/HzVCiwbSl+YfPKCoGnZDXCEsG5bKBi4CMsastUdvp+dejaIJo2rlmbRSaTQ==~-1~-1~-1; bm_sz=B87C0BFD2221B21280A9F9855A098B6C~YAAQ9bEHyuO7Lz6EAQAA2wEaYBEQfGhJU+H3H1HBRj7iHA0VnErcqHawnFDAB8Myh0EXEOOhMVYjHOqzTxi5oQ6hsllnqo86vTkRhvx3K2XYhopzuWQfBaRyM61XLLaZY/+wGJwY/tbiQxnyh6Xq0P5tr0ZkIoSdeTJ3xgo+zjk6Y2zDxKvmumpWuUj4MG0Krv0cQfVAl1t6O9XHwVwZvzKLUjN6DuimsQcJ6GOCLtro6w0xEH1FlG5Ag+RHRMWp5J5+xTiktO1gLpkSbjvAOCEst4BRW5gUr5TsV8tdq79GmCo=~3294009~4407861; sid_guard=ccee2b81e4750e6ea2143ce3da193ada%7C1668059629%7C5184000%7CMon%2C+09-Jan-2023+05%3A53%3A49+GMT; sid_ucp_v1=1.0.0-KGIxYjc3ZDYxNTIzMjJhYzI3M2Q0Y2JiZTMxMzEyM2U0ZjI2NDMxNTYKIAiFiL_Q2NzO8VwQ7aOymwYYswsgDDCP94znBTgHQPQHEAEaA3NnMSIgY2NlZTJiODFlNDc1MGU2ZWEyMTQzY2UzZGExOTNhZGE; ssid_ucp_v1=1.0.0-KGIxYjc3ZDYxNTIzMjJhYzI3M2Q0Y2JiZTMxMzEyM2U0ZjI2NDMxNTYKIAiFiL_Q2NzO8VwQ7aOymwYYswsgDDCP94znBTgHQPQHEAEaA3NnMSIgY2NlZTJiODFlNDc1MGU2ZWEyMTQzY2UzZGExOTNhZGE; csrf_session_id=0a7ac5c360c0dc630815b8a67237aec0; ttwid=1%7CrQEy5vKY3idfcFF58rL18KYh01LCRiHodGmzovQwpJE%7C1668059740%7Caab0656c0e3c3958c824dccc9a675b8db133bcd1a9a051438c3b082bbbfb3012; odin_tt=2784aa62bd7a3ae74c2945c572aef495af422ea5f1986fa055060fa025baa4390a0457874e1e6035be44d4a02eebcbd7fdb4216595287a31bbd1e3ded7ca34c58dd2d4cb1826ce31066177c8a9898811; csrfToken=iZKs2HkKKiW-G77OQgHL0bI_; msToken=qqL_LL1SgBvIGmZAf9LW7luKyQhm7WHYCW0g4OZ7AqZq2CESkHZ0EKfNQoHWxPVn0XXMCo93CAPunyzFH3csp0iEbotTFEsxw0mDDmyxU9RBbHn11Rqywbs0iyDsoT9so0BvyJJaQliJM74=' \
  -H 'origin: https://www.tiktok.com' \
  -H 'referer: https://www.tiktok.com/' \
  -H 'sec-ch-ua: "Google Chrome";v="107", "Chromium";v="107", "Not=A?Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-site' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36' \
  --compressed
```
300kb
https://gist.githubusercontent.com/johndpope/6b4d606bc30e33a091164f6a6ab5a702/raw/d86d4c2b24086eba260b7a331bacb52503c397a1/gistfile1.txt

**Comment by isaackogan** [comment]
> ```shell
> curl 'https://webcast.tiktok.com/webcast/feed/?aid=1988&app_language=en&app_name=tiktok_web&battery_info=0.95&browser_language=en-GB&browser_name=Mozilla&browser_online=true&browser_platform=MacIntel&browser_version=5.0%20%28Macintosh%3B%20Intel%20Mac%20OS%20X%2010_15_7%29%20AppleWebKit%2F537.36%20%28KHTML%2C%20like%20Gecko%29%20Chrome%2F107.0.0.0%20Safari%2F537.36&channel=tiktok_web&channel_id=86&content_type=0&cookie_enabled=true&device_id=7134872442754188802&device_platform=web_pc&focus_state=false&from_page=user&hidden_banner=true&history_len=16&is_fullscreen=false&is_page_visible=true&max_time=0&os=mac&priority_region=AU&referer=&region=AU&req_from=pc_web_suggested_host&screen_height=982&screen_width=1512&tz_name=Australia%2FSydney&webcast_language=en&msToken=qqL_LL1SgBvIGmZAf9LW7luKyQhm7WHYCW0g4OZ7AqZq2CESkHZ0EKfNQoHWxPVn0XXMCo93CAPunyzFH3csp0iEbotTFEsxw0mDDmyxU9RBbHn11Rqywbs0iyDsoT9so0BvyJJaQliJM74=&X-Bogus=DFSzswVuGnJAN9jsS0il1p7TlqSA&_signature=_02B4Z6wo00001W0AuAwAAIDC-dbjA01oJeVtALyAADgw06' \
>   -H 'authority: webcast.tiktok.com' \
>   -H 'accept: */*' \
>   -H 'accept-language: en-GB,en-US;q=0.9,en;q=0.8' \
>   -H 'cookie: _ttp=2CihgY1ygHg9OrojPiipd9WeTgX; passport_csrf_token=98536af196df0de0bda32572ae7939a8; passport_csrf_token_default=98536af196df0de0bda32572ae7939a8; cmpl_token=AgQQAPPdF-RO0ozd1EICPR08-WWxnJOJf4B0YMTRuw; uid_tt=146da7d8c5724930d8e2630a027b2bbd06a20f460f778e5142e6835d110db00b; uid_tt_ss=146da7d8c5724930d8e2630a027b2bbd06a20f460f778e5142e6835d110db00b; sid_tt=ccee2b81e4750e6ea2143ce3da193ada; sessionid=ccee2b81e4750e6ea2143ce3da193ada; sessionid_ss=ccee2b81e4750e6ea2143ce3da193ada; store-idc=alisg; store-country-code=au; store-country-code-src=uid; tt-target-idc=alisg; tt-target-idc-sign=ZFMl3e6uDrGUqsROQcgB29DAcmiO-YJgkcxowQKCyNiaUHklxupyS5-XBZH7GuBp2O8SJEUwCkJGiiqRQ9nx5ra7rSHerFk_TBe-p6IAEwN8-w09AwDg9bnYkCsKNGdDef5LGqmLBOMZQxEX1OGt3zQYP5yDuXhABEjMymldyDcKlB_QNizqZItlWrA6KIvHqkgvVydMlJaSuUIJhw3Mg2i4kXJ55cwFWSxGrTYhXmX3YFF8czGHXVnUcpXaOxQQCnbpLFMRnwvNp9TLM0dfIytDMlJYNhIdMScQgcxXdESChe_7ByO0k_jRHnASylHHKHAbCT5w2rBYRb0AdioksujA4Ibu8mohVEli5UOo5kRFiZ6rvg18rzYmMzwLecAw8TZ9H8siAGiMBbY1Hd9ws8Et8H2v_Vn8scoFUQKFBygrT-Jf8vxPawZm4fce91Y_AlyQQqvDe5Si3Y6IWFT9sWpyF16A8GatYZsgiZLxmFI10QjYrNLuCQFtSkEsfBNY; tt_csrf_token=jEJBLjOD-FwitMB_B-G5WyYM40ZSslJAodFw; _abck=70BC0444C961E9B31B11D5B29C4F0759~-1~YAAQ9bEHyuK7Lz6EAQAA2wEaYAjaX/I4THwHn9lWtNk0TXkCgVXx5+vAqzuqn/Xb4VjsI6nA0EWJGn3wuNVf7OuHQStz/ba2xXaKldHhk/YNeELcPplr6ZLm04qZE+7DyUVotJItN0OqCTyNOjOb3CQUPNJ51jVgsGfQiXj2WOirVh8usVSq/jC0RlMyGIFJpKX6UQ3xMgTO0EmbqkE8WnEn9mnRWCV25Dfs0lRIOMntCtqPhjGCYX3RZMDktBbnSEKECkCxPvM4Q1PpsT7gXAcGa6dliL0vOFeErNF82V4FMFvQQabVGtLKNsb+dGxdQqsBJJGNazbnyQ4emu/HzVCiwbSl+YfPKCoGnZDXCEsG5bKBi4CMsastUdvp+dejaIJo2rlmbRSaTQ==~-1~-1~-1; bm_sz=B87C0BFD2221B21280A9F9855A098B6C~YAAQ9bEHyuO7Lz6EAQAA2wEaYBEQfGhJU+H3H1HBRj7iHA0VnErcqHawnFDAB8Myh0EXEOOhMVYjHOqzTxi5oQ6hsllnqo86vTkRhvx3K2XYhopzuWQfBaRyM61XLLaZY/+wGJwY/tbiQxnyh6Xq0P5tr0ZkIoSdeTJ3xgo+zjk6Y2zDxKvmumpWuUj4MG0Krv0cQfVAl1t6O9XHwVwZvzKLUjN6DuimsQcJ6GOCLtro6w0xEH1FlG5Ag+RHRMWp5J5+xTiktO1gLpkSbjvAOCEst4BRW5gUr5TsV8tdq79GmCo=~3294009~4407861; sid_guard=ccee2b81e4750e6ea2143ce3da193ada%7C1668059629%7C5184000%7CMon%2C+09-Jan-2023+05%3A53%3A49+GMT; sid_ucp_v1=1.0.0-KGIxYjc3ZDYxNTIzMjJhYzI3M2Q0Y2JiZTMxMzEyM2U0ZjI2NDMxNTYKIAiFiL_Q2NzO8VwQ7aOymwYYswsgDDCP94znBTgHQPQHEAEaA3NnMSIgY2NlZTJiODFlNDc1MGU2ZWEyMTQzY2UzZGExOTNhZGE; ssid_ucp_v1=1.0.0-KGIxYjc3ZDYxNTIzMjJhYzI3M2Q0Y2JiZTMxMzEyM2U0ZjI2NDMxNTYKIAiFiL_Q2NzO8VwQ7aOymwYYswsgDDCP94znBTgHQPQHEAEaA3NnMSIgY2NlZTJiODFlNDc1MGU2ZWEyMTQzY2UzZGExOTNhZGE; csrf_session_id=0a7ac5c360c0dc630815b8a67237aec0; ttwid=1%7CrQEy5vKY3idfcFF58rL18KYh01LCRiHodGmzovQwpJE%7C1668059740%7Caab0656c0e3c3958c824dccc9a675b8db133bcd1a9a051438c3b082bbbfb3012; odin_tt=2784aa62bd7a3ae74c2945c572aef495af422ea5f1986fa055060fa025baa4390a0457874e1e6035be44d4a02eebcbd7fdb4216595287a31bbd1e3ded7ca34c58dd2d4cb1826ce31066177c8a9898811; csrfToken=iZKs2HkKKiW-G77OQgHL0bI_; msToken=qqL_LL1SgBvIGmZAf9LW7luKyQhm7WHYCW0g4OZ7AqZq2CESkHZ0EKfNQoHWxPVn0XXMCo93CAPunyzFH3csp0iEbotTFEsxw0mDDmyxU9RBbHn11Rqywbs0iyDsoT9so0BvyJJaQliJM74=' \
>   -H 'origin: https://www.tiktok.com' \
>   -H 'referer: https://www.tiktok.com/' \
>   -H 'sec-ch-ua: "Google Chrome";v="107", "Chromium";v="107", "Not=A?Brand";v="24"' \
>   -H 'sec-ch-ua-mobile: ?0' \
>   -H 'sec-ch-ua-platform: "macOS"' \
>   -H 'sec-fetch-dest: empty' \
>   -H 'sec-fetch-mode: cors' \
>   -H 'sec-fetch-site: same-site' \
>   -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36' \
>   --compressed
> ```
> 
> 300kb https://gist.githubusercontent.com/johndpope/6b4d606bc30e33a091164f6a6ab5a702/raw/d86d4c2b24086eba260b7a331bacb52503c397a1/gistfile1.txt

Is this data the list of streamers you were looking for?
